sdk/storage/azure-storage-file-share/src/samples/java/com/azure/storage/file/share/ReadmeSamples.java

```diff
@@ -89,6 +93,21 @@ public void createShareClientWithConnectionString() {
         // END: readme-sample-createShareClientWithConnectionString
     }
 
+    public void createShareServiceClientWithTokenCredential() {
+        String shareName = "testshare";
+
+        // BEGIN: readme-sample-createShareClientWithTokenCredential
+        String shareURL = String.format("https://%s.file.core.windows.net", ACCOUNT_NAME);
+
+        ShareClient serviceClient = new ShareClientBuilder()
+            .endpoint(shareURL)
+            .credential(tokenCredential)
+            .shareName(shareName)
+            .shareTokenIntent(ShareTokenIntent.BACKUP)
+            .buildClient();
```

Comment:
I would put `sharetokenIntent()` immediately after `credential()` since they are grouped together conceptually and this sample is about that exact grouping.